elide-contrib/elide-dynamic-config-helpers/src/main/resources/elideTableSchema.json

```diff
@@ -0,0 +1,409 @@
+          "description": "Read permission for the dimension.",
+          "type": "string",
+          "default": "Allow All"
+        },
+        "definition": {
+          "title": "Dimension definition",
+          "description": "The definition of the dimension",
+          "type": "string",
+          "default": ""
+        },
+        "tags": {
+          "title": "Dimension tags",
+          "description": "An array of string based tags for dimensions",
+          "type": "array",
+          "uniqueItems": true,
+          "items": {
+            "type": "string"
+          },
+          "default": []
+        }
+      }
+    },
+    "dimension": {
+      "title": "Dimension",
+      "description": "Dimensions represent labels for measures. Dimensions are used to filter and group measures.",
+      "type": "object",
+      "allOf": [
+        { "$ref": "#/definitions/dimensionRef" },
+        {
+          "properties": {
+            "type": {
+              "oneOf":[
+              {"$ref": "#/definitions/enumtype"},
+              {"enum": [
+                "RELATIONSHIP","ID"
+              ]}
+              ],
+              "default": "TEXT"
+            }
+          }
+        }
+      ],
+      "required": [
+        "name",
+        "type",
+        "definition"
+      ]
+    },
+    "timeDimension": {
+      "title": "Time Dimension",
+      "description": "Time Dimensions represent labels for measures. Dimensions are used to filter and group measures.",
+      "type": "object",
+      "allOf": [
+        { "$ref": "#/definitions/dimensionRef" },
+        {
+          "properties": {
+            "type": {
+              "title": "Dimension field type",
+              "description": "The data type of the dimension field",
+              "type": "string",
+              "enum": [
+                "TIME"
+              ],
+              "default": "TIME"
+            },
+            "grains": {
+              "title": "Time Dimension grains",
+              "description": "Time Dimension granularity and Sqls",
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/grain"
+              },
+              "default": []
+            }
+          }
+        }
+      ],
+      "required": [
+        "name",
+        "type",
+        "definition",
+        "grains"
+      ]
+    }
+  },
+  "type": "object",
+  "properties": {
+    "tables": {
+      "title": "Elide Table Models",
+      "description": "Array of elide Table Models",
+      "type": "array",
+      "uniqueItems": true,
+      "items": {
+        "title": "Elide Table Models",
+        "description": "Array of elide Table Models",
+        "type": "object",
+        "properties": {
+          "name": {
+            "title": "Table Model Name",
+            "description": "The name of the model. This will be the same as the POJO class name.",
+            "type": "string",
+            "pattern": "^[A-Z][A-Za-z]*$"
```

Comment:
name and schema I believe can also include numbers in all but the first character.